lib/omnibus/fetchers/git_fetcher.rb

```diff
@@ -193,6 +192,13 @@ def git(command)
 
     def self.resolve_version(ref, source)
       if sha_hash?(ref)
+        # A git server negotiates in terms of refs during the info-refs phase of a fetch.
+        # During upload-pack, the client is not allowed to specify any sha1s in the "wants"
+        # unless the server has publicized them during info-refs.  Hence, the server is allowed
+        # to drop requests to fetch particular sha1s, even if it is an otherwise reachable
+        # commit object.  Only when the service is specifically configured with
+        # uploadpack.allowReachableSHA1InWant is there any guarantee that it consider "naked" wants.
+        log.warn(log_key) { 'A sha1 hash is not a ref name and there is no guarantee that you will be fetching the commits you need.'}
```

Comment:
Based on how we resolve the manifest currently, isn't this going to warn on pretty much every git-based software definition we have?